OutOfSchool/OutOfSchool.WebApi/Services/AverageRatings/AverageRatingService.cs

```diff
@@ -0,0 +1,222 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore.Storage;
+using Nest;
+using NuGet.Packaging;
+using OutOfSchool.ElasticsearchData.Models;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Common.QuartzConstants;
+using OutOfSchool.WebApi.Models;
+using System.Collections.Generic;
+using System.Drawing.Text;
+using System.Net.WebSockets;
+
+namespace OutOfSchool.WebApi.Services.AverageRatings;
+
+public class AverageRatingService : IAverageRatingService
+{
+    private readonly ILogger<AverageRatingService> logger;
+    private readonly IMapper mapper;
+    private readonly IAverageRatingRepository averageRatingRepository;
+    private readonly IRatingService ratingService;
+    private readonly IWorkshopRepository workshopRepository;
+    private readonly IRatingRepository ratingRepository;
```

Comment:
replace this type with `IEntityRepository<long, Rating>` here and in constructor, the default implementation will be created and injected automatically

Reply:
Fixed.